### ISSUE TITLE
Warn of a higher grade road only connecting to lower grade roads

### ISSUE THREAD

**Issue by jidanni** [openstreetmap/iD#6892]
If one tries to save an isolated road in the middle of nowhere, iD will warn that it is unconnected to the road network.

That's good. But here is another situation iD could warn about:
- A higher grade road only connecting to lower grade roads.

![19029-1](https://user-images.githubusercontent.com/1230959/65826905-fb57a700-e2be-11e9-8c26-075c0bf67160.jpg)

- A higher grade road only connecting to lower grade roads.

**Comment by quincylvania** [comment]
@jidanni Can you provide a link to a location you're seeing this?

I'm not sure this is always an error or something we need to bother iD mappers about, but perhaps I could be convinced.

**Comment by jidanni** [comment]
(No link because I already hastily fixed that road already.)
Perhaps I could be convinced such items are not mistakes: provide one shopping center (campground, etc., even suburb) whose bigger road(s) inside are connected to the outside world by a smaller/poorer road.

**Comment by quincylvania** [comment]
@jidanni Yeah even if this warning would be valid it doesn't seem like a common or important issue for OSM right now. I think let's spend our development time elsewhere.

**Comment by jidanni** [comment]
(Actually, the current check,

- Is this road not connected to any other roads?

Is just a special case of the better check,

- Is this road not connected to any other equal or higher class roads?

And in fact the same concept can be applied to any hierarchy,

- roads, paths, trails
- streams
- bodies of water (ocean > sea > bay)
- railroads
- telephone, power, cable TV, gas. oil pipeline etc. networks
- cell towers
- political and cadastral boundaries (country > state > county ...)

Not only for lines and areas, but more generally:

- Is this chain store missing which regional group it belongs to?

And in fact depending on what part of the world we are in, we can enforce

- this District Office must belong to a Regional Office. You can't just
  directly fill in the Headquarters or leave it blank, without getting a
  warning.

And in the very rare case one is e.g., actually
- Creating a new independent country
- Building an island's first expressway

One can ignore the warnings.)

**Comment by quincylvania** [comment]
@jidanni Those are interesting ideas that might be useful for another tool, but again, I don't think this kind of thing is worth bothering iD mappers over right now.

Also, please refrain from changing the scope of issues after they've been closed. It creates confusion and is not an effective way to have an idea considered.
